Odata-docs/performance/mark-edm-model-immutable.md

```diff
@@ -0,0 +1,53 @@
+---
+title:  "Mark the IEdmModel as immutable"
+description: A guide on how to improve performance by marking the IEdmModel as immutable.
+date:   2023-08-11
+ms.date: 9/5/2022
+author: habbes
+ms.author: clhabins
+---
+
+# Mark the `IEdmModel` as immutable
+
+**Applies To**:[!INCLUDE[appliesto-webapi](../includes/appliesto-webapi-v8.md)][!INCLUDE[appliesto-webapi](../includes/appliesto-webapi-v7.md)][!INCLUDE[appliesto-odatalib](../includes/appliesto-odatalib-v7.md)]
+
+OData libraries may perform multiple lookups in the `IEdmModel` when processing a request. Some of these lookups may be expensive, especially for large models with a lot of schema elements.
+We have some optimizations in place to reduce the performance overhead of these internal lookups, but some of these optimizations are only applicable if the library knows that the EDM model
+will not change. You can enable these optimization by explicitly marking your EDM as immutable by calling the `MarkAsImmutable` method, which is an extension method in the `Microsoft.OData.Edm` namespace.
+
+```csharp
+using Microsoft.OData.Edm;
+
+IEdmModel GetEdmModel()
+{
+    var modelBuilder = new ODataConventionModelBuilder();
+    modelBuilder.EntitySet<Person>("People");
+    IEdmModel model = modelBuilder.GetEdmModel();
+    model.MarkAsImmutable();
+    return model;
+}
+```
+
+It is recommended to mark the `IEdmModel` as immutable since most applications do not modify the model after it has been constructed.
```

Comment:
```suggestion
It is recommended to mark the Edm model as immutable since most applications do not modify the model after it has been constructed.
```